pinot-common/src/main/java/org/apache/pinot/common/function/scalar/MapFunctions.java

```diff
@@ -0,0 +1,213 @@
+/**
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+package org.apache.pinot.common.function.scalar;
+
+import com.fasterxml.jackson.core.type.TypeReference;
+import com.fasterxml.jackson.databind.ObjectMapper;
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.LinkedHashMap;
+import java.util.List;
+import java.util.Map;
+import org.apache.pinot.spi.annotations.ScalarFunction;
+
+public class MapFunctions {
+    private MapFunctions() { }
+
+    public static final ObjectMapper oM = new ObjectMapper();
+
+    /**
+     * Internal Helper Function to convert Json String Values to Java Maps
+     *
+     * @param jsonString
+     * @return converted Java Map from JSON String
+     */
+    public static LinkedHashMap stringToMap(String jsonString) {
+        // input validation for ObjectMapper.readValue();
+        if (jsonString == null) {
+            return null;
+        }
+
+        LinkedHashMap<Object, Object> convertedMap = null;
+        try {
+            // object --> java map with objectMapper
+            convertedMap =
+                oM.readValue(jsonString, new TypeReference<LinkedHashMap<Object, Object>>() { });
+        } catch (Exception e) {
+            throw new RuntimeException("Unknown issue: " + jsonString);
+        }
+
+        return convertedMap;
+    }
+
+    /**
+     * Scalar listMapEntries UDF that lists all key-value entries from a String representation of JSON
+     * Map
+     *
+     * @param jsonString
+     * @return string array of all key-value pairs from JSON map in data
+     */
+    @ScalarFunction
+    public static String[] listMapEntries(String jsonString) {
+        // handle empty string case
+        if (jsonString == null) {
```

Comment:
If you want this function to be able to take `null` values as input, you'll need to add `nullableParameters = true` to the `@ScalarFunction` annotation. By default, scalar functions are assumed to be null "intolerant", so if any argument to the function is `null`, the value `null` will be returned without even invoking the function (see [FunctionInvoker](https://github.com/apache/pinot/blob/56ccbc3d5f04f0d62041c32d314b814c1b7cab4f/pinot-common/src/main/java/org/apache/pinot/common/function/FunctionInvoker.java#L33)). Same goes for the other scalar functions added here.